### DISCUSSION TITLE
Apiable Migration

### DISCUSSION THREAD

**Discussion by alexander0042** [Pirate-Weather/pirateweather#60] [General]
## Centralized discussion for Apiable related improvements!

While one of the goals for this API is to do all the weather data processing from model to (HTTP) method, I don't have the background or the desire to handle the sign up, authentication, or billing side of things. [Apiable.io](https://apiable.io) is a fantastic startup that's devoted to doing exactly this, so a perfect fit for Pirate Weather. Since they're always improving their platform, I wanted to centralize discussion here to pass ideas along to their developers

**Comment by alexander0042** [comment]
Referencing issues:
- #52 
- #57 
- #55 
- #54 
- #59

**Comment by cloneofghosts** [comment]
I think this issue also applies here https://github.com/alexander0042/pirate-weather-ha/issues/105 and this one as well https://github.com/alexander0042/pirateweather/issues/42 sort of relates

What sort of options does APIable give you for the plans table? Is it only true/false and nothing else? It would be nice to include the number of requests per month, number of calls per second and historical calls per month just as examples. One thing that should be added is which tiers give you commercial support which it doesn't say at the moment. I would assume the $2/month plan does but it doesn't say.

Posted this in my thread but here's other pricing pages as examples:
https://developer.accuweather.com/packages
https://openweathermap.org/price
https://www.weatherbit.io/pricing
https://www.weatherapi.com/pricing.aspx
https://www.visualcrossing.com/weather-data-editions

**Reply by cloneofghosts** [reply]
Here's some suggestions to pass along that I can think of currently:

- More options for the plans table.
- Make Organization not required on signup. I know you can just put Personal or anything in there but it would be nice if it wasn't required.
- There seems to be no way to edit your own profile. The only options in the dropdown are to logout or change team.
- The only option on a team is to invite people to the team. I see no option to change the name of the team at all after creating it.
- There is no way to edit your subscription other than cancelling it. If someone wanted to change tiers they'd have to cancel their subscription and subscribe again. I don't have a paid plan so I cannot confirm if you can change billing details at all.
- The menu shows your name and the team name which can lead to some of the text being off the screen. Team name could go in the dropdown instead
![image](https://github.com/alexander0042/pirateweather/assets/10248058/eb26062e-de64-40f3-9e50-a611ca386318)
- Is it possible to have it redirect you to the plans page after you activate your account? Currently it takes you to the homepage and you have to browse back to the plans page to finish the subscription. More of an annoyance than anything.

I'm not sure how well discussion notifications work so I'll ping you in case it doesn't send you a notification for whatever reason. @alexander0042

**Comment by alexander0042** [comment]
Two notes on this topic! One is that the new kong backend provides subscription usage, so that should solve the usage tracking issue (and better align with how Dark Sky did things). Also, Apiable added a new button to access receipts on the subscription page- it's not as convenient as e-mail, but provides a decent starting point/ temporary workaround.

**Reply by pdxvr** [reply]
Are we able to track usage yet?
